modules/testing/run/bucket.tf

```diff
@@ -0,0 +1,23 @@
+resource "aws_s3_bucket" "this" {
+  bucket        = random_pet.run.id
+  force_destroy = true
+}
+
+resource "aws_s3_access_point" "this" {
+  count  = var.enable_access_point ? 1 : 0
+  bucket = aws_s3_bucket.this.id
+
+  name = random_pet.run.id
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
+  count  = length(aws_kms_key.this)
+  bucket = aws_s3_bucket.this.id
+
+  rule {
+    apply_server_side_encryption_by_default {
```

Comment:
so does this not get applied if the aws_kms_key doesn't get created?

Reply:
Correct, it is gated by `count`, which in turn is gated on whether user provided us with a policy.